### DISCUSSION TITLE
Is RulesEngine.ExecuteAllRulesAsync thread safe?

### DISCUSSION THREAD

**Discussion by samholder** [microsoft/RulesEngine#438] [Q&A]
General question really. As in the title I just want to know if RulesEngine.ExecuteAllRulesAsync is thread safe?

So can I initialise one instance of the rules engine with all my workflows with all my rules in them and then safely execute the rules against one particular workflow without worrying about any other potential executions against other workflows?

**Comment by abbasc52** [comment]
RulesEngine is thread safe. It is intended to be used anywhere including API where thread safety is a must.
